logstash-core/src/main/java/org/logstash/common/io/ByteBufferPageIO.java

```diff
@@ -41,208 +24,41 @@ public ByteBufferPageIO(int capacity) throws IOException {
-                String.format("seqNum=%d is > maxSeqNum=%d", seqNum, maxSeqNum());
-
-        List<byte[]> elements = new ArrayList<>();
-        List<Long> seqNums = new ArrayList<>();
-
-        int offset = this.offsetMap.get((int)(seqNum - this.minSeqNum));
-
-        this.buffer.position(offset);
-
-        for (int i = 0; i < limit; i++) {
-            long readSeqNum = this.buffer.getLong();
-
-            assert readSeqNum == (seqNum + i) :
-                    String.format("unmatched seqNum=%d to readSeqNum=%d", seqNum + i, readSeqNum);
-
-            int readLength = this.buffer.getInt();
-            byte[] readBytes = new byte[readLength];
-            this.buffer.get(readBytes);
-            int checksum = this.buffer.getInt();
-            int computedChecksum = checksum(readBytes);
-            if (computedChecksum != checksum) {
-                throw new IOException(String.format("computed checksum=%d != checksum for file=%d", computedChecksum, checksum));
-            }
-
-            elements.add(readBytes);
-            seqNums.add(readSeqNum);
-
-            if (seqNum + i >= maxSeqNum()) {
-                break;
-            }
-        }
-
-        return new SequencedList<>(elements, seqNums);
     }
 
     @Override
-    public void deactivate() {
-        // nothing to do
-    }
+    public void deactivate() { /* nothing */ }
 
     @Override
-    public void activate() {
-        // nothing to do
-    }
+    public void activate() { /* niet */ }
 
     @Override
-    public void ensurePersisted() {
-        // nothing to do
-    }
+    public void ensurePersisted() { /* nada */ }
 
     @Override
-    public void purge() throws IOException {
-        // do nothing
-    }
+    public void purge() { /* zilch */ }
 
     @Override
-    public void close() throws IOException {
-        // TODO: not sure if we need to do something here since in-memory pages are ephemeral
-    }
+    public void close() { /* don't look here */ }
 
-    private int checksum(byte[] bytes) {
-        checkSummer.reset();
-        checkSummer.update(bytes, 0, bytes.length);
-        return (int) checkSummer.getValue();
-    }
-
-    // TODO: static method for tests - should refactor
-    public static int _persistedByteCount(int byteCount) {
-        return SEQNUM_SIZE + LENGTH_SIZE + byteCount + CHECKSUM_SIZE;
-    }
 
     @Override
-    public int persistedByteCount(int byteCount) {
-        return ByteBufferPageIO._persistedByteCount(byteCount);
-    }
-
-    private long maxSeqNum() {
-        return this.minSeqNum + this.elementCount - 1;
-    }
-
+    protected ByteBuffer getBuffer() { return this.buffer; }
 
     // below public methods only used by tests
 
-    public int getWritePosition() {
-        return this.head;
-    }
+    // TODO: static method for tests - should refactor
+    public static int _persistedByteCount(int byteCount) { return SEQNUM_SIZE + LENGTH_SIZE + byteCount + CHECKSUM_SIZE; }
```

Comment:
so - at this point it cannot "easily" be other than public because there are many usage outside that package ... we should defer that to another issue/pr as this is really out of scope for this pr. ok?